src/components/external-video-player/index.js

```diff
@@ -0,0 +1,318 @@
+    this.handleVolumeChange = this.handleVolumeChange.bind(this);
+    this.handleOnBuffer = this.handleOnBuffer.bind(this);
+    this.handleOnBufferEnd = this.handleOnBufferEnd.bind(this);
+        
+    this.orchestrator = this.orchestrator.bind(this);
+    this.autoPlayBlockDetected = this.autoPlayBlockDetected.bind(this);
+   
+  }
+
+  autoPlayBlockDetected() {
+    this.setState({ autoPlayBlocked: true });
+  }
+
+  handleFirstPlay() {
+    const { hasPlayedBefore } = this;
+
+    if (!hasPlayedBefore) {
+      this.hasPlayedBefore = true;
+
+      this.setState({ autoPlayBlocked: false });
+
+      if (this.autoPlayTimeout) {
+        clearTimeout(this.autoPlayTimeout);  
+      }
+          
+    }
+  }
+
+  getCurrentTime() {
+    if (this.player && this.player.getCurrentTime) {
+      return Math.round(this.player.getCurrentTime());
+    }
+  }
+
+  
+  setPlaybackRate() {
+
+    const { primaryPlaybackRate } = this.props;
+
+    // Rate depends on primary rate player
+    const rate = primaryPlaybackRate * this.lastEventPlaybackRate;
+
+    const currentRate = this.state.playbackRate;
+
+    logger.debug(`external_video: setPlaybackRate current=${currentRate} primary=${primaryPlaybackRate} lastEventPlaybackRate=${this.lastEventPlaybackRate} rate=${rate}`);
+
+    if (currentRate === rate) {
+      return;
+    }
+
+    this.setState({ playbackRate: rate });
+
+  }
+
+  handleOnReady() {
+    const { hasPlayedBefore, playerIsReady } = this;
+
+    if (hasPlayedBefore || playerIsReady) {
+      return;
+    }
+
+    this.playerIsReady = true;
+    this.handleFirstPlay();
+
+    const { onPlayerReady } = this.props;
+  
+    if (onPlayerReady) onPlayerReady(ID.EXTERNAL_VIDEOS, this);
+
+      
+  }
+
+  handleOnPlay() {
+    const { playing } = this.state;
+
+    if (!playing && this.primaryPlayerPlaying) {
+        this.setState({ playing: true });
+        this.handleFirstPlay();
+    }
+  }
+
+  handleOnPause() {
+    const { playing } = this.state;
+
+    if (playing) {
+      this.setState({ playing: false });
+      this.handleFirstPlay();
+    }
+  }
+
+  handleOnBuffer() {
+    this.buffering = true;
+  }
+
+  handleOnBufferEnd() {
+    this.buffering = false;
+  }
+
+  handleVolumeChange = (value, isMuted) => {
+    this.setState({ volume: parseFloat(value)});
+    this.setState({ muted: isMuted});
+  }
+
+  
+  seekTo(time) {
+    const { player } = this;
+    
+    if (!player) {
+      //return logger.error('No player on seek');
+      return;
+    }
+
+    // Seek if viewer has drifted too far away from presenter
+    if (Math.abs(this.getCurrentTime() - time) > SYNC_INTERVAL_SECOND * 0.75) {
+      player.seekTo(time, true);
+    }
+  }
+
+  componentDidMount () {
+    this.timer = setInterval(() => this.orchestrator(), ORCHESTRATOR_INTERVAL_MILLISECOND);
+  }
+
+  componentWillUnmount () {
+    clearInterval(this.timer);
+  }
+
+  orchestrator () {
+    const { events, active, primaryPlaybackRate } = this.props;
```

Comment:
We need to bring the timestamp update function here.
`const {getCurrentPlayerTime } = this.props;`